### DISCUSSION TITLE
Unpopulated eager'd relationship on browser

### DISCUSSION THREAD

**Discussion by kennethsato** [mikro-orm/mikro-orm#3049] [Q&A]
Hello!
I've got a entity called "Insercao", which has a child "Comercial", as follows:

```
@Entity({ tableName: 'insercao' })
export default class Insercao {

.... props....

  @ManyToOne(() => Comercial, { eager: true })
  comercial: Comercial | null;
```

And I'm trying to do a simple findById on it, populating Comercial entity:

  ```
public async findById(id: number): Promise<Insercao | null> {
    const insercao = await this.ormRepository.findOne(
      { id },
      {
        populate: ['emissora', 'cliente', 'comercial', 'comercial.midia'],
      },
    );
    return insercao;
  }
```

Unfortunatelly, the Comercial entity is not being populated as expected when I call it by a browser (tested Chrome and Edge) as you can see on the following screenshot
![unpopulated comercial](https://user-images.githubusercontent.com/33872589/164712190-7c40b320-d8b8-4941-8058-8cea3a2186b3.png)
"

But when I do the exact same request on Insomnia, I get it populated as following:
![populated comercial](https://user-images.githubusercontent.com/33872589/164712525-1f50bf8b-c27b-4d9f-84bf-094a93022579.png)

You can also see that other relationships are populated on browser.

I tried adding "cache: false" on my repository and I tried specifying fields, but got no success. 

I'm not really sure if it's a Mikro-ORM discussion, once it's fine on Insomnia and I also logged the response on backend and it's also populated, but I don't know what else could it be.

Could you please help me?

**Comment by B4nan** [comment]
Its probably not marked as populated, maybe you had that entity loaded in the context before? This also happens if you initialize things via `wrap(e).init()`. To verify this, you could try `disableIdentityMap: true` in the find options.

You can also mark it explicitly as populated via via `wrap(ent.comercial).populated()`.

**Reply by kennethsato** [reply]
Thank you for the answer @B4nan 

I tried setting  disableIdentityMap: true on my repository method
```
  public async findById(id: number): Promise<Insercao | null> {
    const insercao = await this.ormRepository.findOne(
      { id },
      {
        disableIdentityMap: true,
        populate: ['emissora', 'cliente', 'comercial', 'comercial.midia'],
      },
    );
    return insercao;
  }
```

Now it's not populated on brower neither on Insomnia.

About it being loaded before:
I think I'm loading that entity on my first get. When I list all Insercao itens.
And I get the "error" when I try to update one of them, setting a Comercial to an Insercao.

And finally, I'm sorry, but I don't get where these wrap init and populated supposed to be.

**Reply by B4nan** [reply]
Oh wait a second, so it works _sometimes_. This means you probably don't have request contexts set up correctly, or your async context is broken. Are you on v5? If not, upgrade, this is validated there.

> And finally, I'm sorry, but I don't get where these wrap init and populated supposed to be.

```ts
  public async findById(id: number): Promise<Insercao | null> {
    const insercao = await this.ormRepository.findOne(
      { id },
      {
        disableIdentityMap: true,
        populate: ['emissora', 'cliente', 'comercial', 'comercial.midia'],
      },
    );
    wrap(insercao.comercial).populated();
    return insercao;
  }
```

But if it works sometimes, its definitely broken context issue, as that leads to instable API responses (= things can be sometimes populated and sometimes not).

**Reply by kennethsato** [reply]
Oh, I get it.
And you're right. I'm still on v4
```
    "@mikro-orm/core": "4.5.9",
    "@mikro-orm/mongodb": "4.5.9",
    "@mikro-orm/mysql": "4.5.9",
    "@mikro-orm/reflection": "4.5.9",
```

I'll updgrade the package and I'll let you know!

And about it working sometimes: Yes. It does works, but only when I do a request using Insomnia API Client

Thanks a lot for the help so far!!

**Reply by B4nan** [reply]
> but only when I do a request using Insomnia API Client

I really doubt it's like that. What probably happens is that the first request is not working the way you expect, and after that you observe via insomnia and see it works fine. If you use the same context for both requests, that is what you get. If you would to the first request via insomnia, it would look the same as in the browser probably.

**Reply by kennethsato** [reply]
I've updated all four mikro-orm packages to "^5.1.2".
And unfortunatelly it's still the same result.

If I set disableIdentityMap, I always get unpopulated results.

I tried to use "wrap(insercao.comercial).populated()" suggestion, but wrap(insercao.comercial) is null when requested by browser.
And I get unpopulated "comercial" requesting via Insomnia

I was also able to get unpopulated comercial on Insomnia. It seems to be when I do a request for the first time on an "insercao".
But I do not get it populated when I do a second request for the same insercao on browser.

**Reply by kennethsato** [reply]
Today I did the following test:

Here's my current Controller:

```
  public async update(request: Request, response: Response): Promise<Response> {
    const { id } = request.params;
    const { statusmapa, duracao, comercial_id, statusmaterial, desvincular } =
      request.body;
    const usuario_id = await Usuario.decodedId(request.headers.authorization);
    const updateInsercao = container.resolve(UpdateInsercaoService);
    const insercao = await updateInsercao.execute({
      id,
      statusmapa,
      duracao,
      comercial_id,
      statusmaterial,
      desvincular,
      usuario_id,
    });
    console.log(insercao)
    return response.json(insercao);
  }
```

I've realized that insercao.comercial was populated on my console.log.
So, it means this issue is probably not related to Mikro-ORM.
Then I tried to change my return line to:

`return response.json({insercao: insercao, comercial: insercao.comercial})`

And now my comercial property is populated (although insercao.comercial is still only the FK).

It's not the perfect solution, but it's a workaround.

Any other suggestions and help is appreciated.
